master/getting-started/openstack/upgrade/index.md

```diff
@@ -0,0 +1,67 @@
+{: .alert .alert-danger}
+
+## Service impact
+
+During the upgrade, **all VMs will continue to function normally**:
+there should be no impact on the data plane. However, control plane
+traffic may fail at different points throughout the upgrade.
+
+Generally, users should be prevented from creating or updating virtual
+machines during this procedure, as these actions will fail. VM deletion
+*may* succeed, but will likely be delayed until the end of the upgrade.
+
+For this reason, we highly recommend planning a maintenance window for
+the upgrade. During this window, you should disable all user API access
+to your OpenStack deployment.
+
+
+## Before you begin
+
+- You must first [upgrade](/v2.6/getting-started/openstack/upgrade) 
+  to {{site.prodname}} [v2.6.x](https://github.com/projectcalico/calico/releases) 
+  before you can upgrade to {{site.prodname}} 
+  {{site.data.versions[page.version].first.title}}. 
+
+## Upgrade steps
+
+If you use calicoctl to add Calico policies in addition to those that are derived automatically (by our Neutron driver) from OpenStack security groups - or more generally to configure any Calico data other than the data that is generated by the Calico Neutron driver - you will need to follow the steps marked OPTIONAL below. Otherwise you can skip those OPTIONAL steps.
+
+1. Set up an etcdv3 cluster, if you don't already have etcdv3 capability, either by [upgrading the etcd software of your existing etcdv2 cluster](https://coreos.com/etcd/docs/latest/upgrades/upgrade_3_0.html), or by [installing an etcdv3 cluster on new servers](https://coreos.com/etcd/docs/latest/op-guide/clustering.html).
+  
+   > **Tip**: etcd v3.x still supports the etcd v2 API, as used by {{site.prodname}} v2.6.x.
+   > However, {{site.prodname}} v3.0.0+ does not support the etcd v2 API.  The upgrade steps below
+   > move the data from the etcd v2 API to the etcd v3 API.  This requires an etcd v3.x server.
+   {: .alert .alert-success}
+  
+1. **OPTIONAL Set the version flag.**  On the control node:
+   ```
+   etcdctl set /calico/v1/config/CalicoVersion v2.6.5-7
+   ```
+   > calico-upgrade will not operate on a etcdv2 datastore without this key set.
+   {: .alert .alert-danger}
+   
+1. **[OPTIONAL Install and configure calico-upgrade](/{{page.version}}/getting-started/openstack/upgrade/setup)** 
+
+1. **[OPTIONAL Test the data migration and check for errors](/{{page.version}}/getting-started/openstack/upgrade/test)**
+
+1. **[OPTIONAL Migrate {{site.prodname}} data](/{{page.version}}/getting-started/openstack/upgrade/migrate)** 
+
+1. **[Upgrade {{site.prodname}} packages](/{{page.version}}/getting-started/openstack/upgrade/upgrade)**, first on each compute nodes, then on the controllers. 
+
+## Convert offline data
+
+Finally, if you have any calico resource manifests stored offline (e.g. files checked into code management systems), you should update them to the new API using the conversion tool:
```

Comment:
This I understood to be not so much of a last step as a step that relates strictly to those who have used the "integration guide" method of installation. 

The upgrade for those using the "integration guide" method is the section that you can see at the bottom here /master/getting-started/kubernetes/upgrade/upgrade where it says docs are coming soon... 

I am not sure if there is anything equivalent for OpenStack.

Reply:
There isn't an equivalent integration guide install in openstack.  

However, if you were using calico to lock down the hosts and had host endpoint objects and host protection policy on your Openstack cluster, its not inconceivable that you might have that config checked into a git repo (say).   Those manifests would need updating with the calicoctl tool.